src/P7-PSEngine/Data/PSengineDB.cs

```diff
@@ -5,23 +5,153 @@ namespace P7_PSEngine.Data
 {
     public class PSengineDB : DbContext
     {
-        public PSengineDB(DbContextOptions options)
+        public PSengineDB()
+        {
+
+        }
+
+        public PSengineDB(DbContextOptions<PSengineDB> options)
         : base(options) { }
 
         public DbSet<Todo> Todos => Set<Todo>();
         public DbSet<User> Users => Set<User>();
         public DbSet<FileInformation> FileInformation => Set<FileInformation>();
-        //public DbSet<IndexInformation> IndexInformations => Set<IndexInformation>();
-
-        //protected override void OnModelCreating(ModelBuilder modelBuilder)
-        //{
-        //    modelBuilder.Entity<FileInformation>(model =>
-        //    {
-        //        model.ToTable("file_information");
-        //        model.HasMany(p => p.IndexInformations).WithOne(p => p.FileInformation);
-        //    });
-        //}
+        public DbSet<InvertedIndex> InvertedIndex => Set<InvertedIndex>();
+        public DbSet<TermInformation> TermInformations => Set<TermInformation>();
         public DbSet<CloudService> CloudService => Set<CloudService>();
+
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+
+            // FileInformation entity
+            modelBuilder.Entity<FileInformation>(entity =>
+            {
+                // Primary key (composity key of FileId and UserId)
+                entity.HasKey(e => new { e.FileId, e.UserId });
```

Comment:
Filers key inde i databasen skal også indeholde cloudservice ID de er forbundet til, for at fjerne chancen for duplicate keys